pymdp/maths.py

```diff
@@ -12,6 +12,13 @@
 
 MINVAL = jnp.finfo(float).eps
 
+# --- Toggle for information-gain formulation ---------------------------------
+# Set to ``True`` to use exact information-gain over parameters instead of the legacy
+# spm_wnorm heuristic.  Changing this single line lets you switch behaviour
+# globally without touching the rest of the code-base.
+USE_EXACT_PARAM_INFO_GAIN = True  # ⇦ CHANGE THIS LINE TO ``True`` FOR THE FIX
```

Comment:
Instead of having this toggle be a global variable defined at the top of the function, I suggest it is an argument to `spm_wnorm(..., use_exact=True)` that defaults to True. This way we can easily functionally change between the exact vs. approximate version without having to change a global variable defined within a file